my-app/src/DataStore.jsx

```diff
@@ -437,6 +448,40 @@ class DataStore {
     return Promise.resolve(Array.from(this._persons).filter(obj => obj[1].name === name).map(obj=>obj[1]));
   }
 
+  getPersonByEmail(email, callback){
+    /**
+     * Gets the person with an email
+     * @param email {string} email to search by
+     * @return {Promise} promise for a {Person}
+     *          with the given email
+     */
+    if (this._persons.keys.length == 0) {
+      this.firestore.collection("persons")
+      .onSnapshot({/*config object*/}, (querySnapshot)=> {
```

Comment:
Do you want to use `onSnapShot()` or just `get()` here? Not sure why you need to listen to the database continually.

Reply:
yeah, that's better. Thanks!